controllers/appvolumereplication_controller.go

```diff
@@ -0,0 +1,337 @@
+	if placement != nil && placement.PlacementRef != nil {
+		plRef := placement.PlacementRef
+		if plRef.Namespace == "" {
+			plRef.Namespace = namespace
+		}
+
+		placementRule := &plrv1.PlacementRule{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: plRef.Name, Namespace: plRef.Namespace}, placementRule)
+
+		if err != nil {
+			return "", err
+		}
+
+		if len(placementRule.Status.Decisions) == 0 {
+			return "", fmt.Errorf("AVR PlacementRule has no decisions. Placement rule name %s", plRef.Name)
+		}
+
+		for _, decision := range placementRule.Status.Decisions {
+			// select the secondary cluster
+			if primaryCluster != decision.ClusterName {
+				secondaryCluster = decision.ClusterName
+
+				break
+			}
+		}
+	}
+
+	return secondaryCluster, nil
+}
+
+func (r *AppVolumeReplicationReconciler) createOrUpdateManifestWork(namespace string, primaryCluster string, secondaryCluster string) error {
+
+	manifestWorkName := fmt.Sprintf("%s-manifestwork", primaryCluster)
+	r.Log.Info("Looking for existing ManifestWork", "ManifestWorkName", manifestWorkName)
+	manifestWork, err := r.createManifestWork(primaryCluster, secondaryCluster)
+	if err != nil {
+		return err
+	}
+	found := &ocmworkv1.ManifestWork{}
+	if err := r.Client.Get(context.TODO(),
+		types.NamespacedName{Name: manifestWorkName, Namespace: namespace}, found); err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Creating ManifestWork", "ManifestWork", manifestWork)
+			if err := r.Client.Create(context.TODO(), manifestWork); err != nil {
+				return err
+			}
+		} else {
+			r.Log.Error(err, "Could not fetch ManifestWork", "ManifestWorkName", manifestWorkName)
+			return err
+		}
+	} else if !reflect.DeepEqual(found.Spec, manifestWork.Spec) {
+		manifestWork.Spec.DeepCopyInto(&found.Spec)
+		err := r.Client.Update(context.TODO(), found)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *AppVolumeReplicationReconciler) createManifestWork(
+	primaryCluster string, secondaryCluster string) (*ocmworkv1.ManifestWork, error) {
+
+	vrgClientManifest, err := r.createVRGClientManifest(primaryCluster, secondaryCluster)
+	if err != nil {
+		r.Log.Error(err, "Could not create VRG manifest.")
+		return nil, err
+	}
+
+	manifestwork := &ocmworkv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-VGR-manifestwork", primaryCluster),
+			Namespace: primaryCluster, Labels: map[string]string{"app": "VRG"}},
+		Spec: ocmworkv1.ManifestWorkSpec{
+			Workload: ocmworkv1.ManifestsTemplate{
+				Manifests: []ocmworkv1.Manifest{
+					*vrgClientManifest,
+				},
+			},
+		},
+	}
+	return manifestwork, nil
+}
+
+func (r *AppVolumeReplicationReconciler) createVRGClientManifest(
+	primaryCluster string, secondaryCluster string) (*ocmworkv1.Manifest, error) {
+
+	vrgClientJSON := []byte(fmt.Sprintf(`
+	{
+		"apiVersion": "ramendr.openshift.io/v1alpha1",
+		"kind": "VolumeReplicationGroup",
+		"metadata": {
+		   "name": "%s"
```

Comment:
We want VRG in the same namespace as the Subscription deployables, this would place it in the default NS?

Reply:
Agreed.